### DISCUSSION TITLE
The in-memory storage option with Kubernetes CRD backend

### DISCUSSION THREAD

**Discussion by dmeytin** [Apicurio/apicurio-registry#2324] [Ideas]
With in-memory storage option, when registry instance is drained or restarted the state with all artifacts will be lost.
The proposed optimization: let's define a CRD for the artifact that will be consumed by apicurio operator.
When a new instance of the registry is loaded, it will initialize its state from the Kubernetes resources.

I would happy to hear your opinion on that

**Comment by EricWittmann** [comment]
We actually have an operator that I think does exactly this (or something very similar).  Check this out:

https://github.com/Apicurio/apicurio-registry-content-sync-operator

Perhaps that can be leveraged to accomplish what you suggest?

@jsenko thoughts?

**Reply by dmeytin** [reply]
Thanks a lot, Eric.
If this is the case, why in memory mode is insufficient for production whilst all definitions are persistent as CRDs?

**Reply by EricWittmann** [reply]
Yeah I guess if deploying the in-memory only version of registry **and** the sync operator, then the source of truth becomes the custom resources and therefore the in-memory version of registry is acceptable for production.

Note that there is an option to switch the UI to "read only" mode, which might be appropriate for this use-case.
